pygments/lexers/procfile.py

```diff
@@ -0,0 +1,40 @@
+"""
+    pygments.lexers.procfile
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for Procfile file format.
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, bygroups
+from pygments.token import Name, Number, String, Text, Punctuation
+
+__all__ = ["ProcfileLexer"]
+
+
+class ProcfileLexer(RegexLexer):
+    """
+    Lexer for Procfile file format.
+
+    The format is used to run processus on Heroku or is used by Foreman or
+    Honcho tools.
+    For more information about the definition of the format, see:
+    https://devcenter.heroku.com/articles/procfile#procfile-format
+    """
+    name = 'Procfile'
+    aliases = ['procfile']
+    filenames = ['Procfile']
+
+    tokens = {
+        'root': [
+            (r'^([a-z]+)(:)', bygroups(Name.Label, Punctuation)),
+            (r'"[^"]*"', String),
+            (r"'[^']*'", String),
+            (r'[0-9.]+', Number),
+            (r'\$[a-zA-Z_][\w]*', Name.Variable),
+            (r'(\w+)(=)(\w+)', bygroups(Name.Variable, Punctuation, String)),
+            (r'([\w\s\-\./])', Text),
```

Comment:
I think you want a `+` in here. Right now this matches each individual letter as a single token as you can see in the output, and I doubt that's what you're after. Additionally, this matches whitespace -- I'd suggest a separate rule for whitespace (and using the right token type for whitespace.) In that case you need to match whitespace first because you have a dot in this rule (which I hope is intentional :) )

Reply:
There are 2 patterns: one for `Whitespace` and one for `Text`. These two patterns end with `+`.
Is it good for you?